src-graphql/client.js

```diff
@@ -22,15 +22,61 @@ export default class Client {
     });
   }
 
+  fetchAll(paginatedModels) {
+    return this.graphQLClient.fetchNextPage(paginatedModels).then(({model}) => {
+      // Until we know how hasNextPage will be exposed, we query until the result is empty
+      if (model.length === 0) {
+        return paginatedModels;
+      }
+
+      paginatedModels.push(...model);
+
+      return this.fetchAll(paginatedModels);
+    });
+  }
+
   fetchAllProducts(query = productConnectionQuery()) {
-    return this.graphQLClient.send(query(this.graphQLClient)).then((response) => {
-      return response.model.shop.products;
+    return this.graphQLClient.send(query(this.graphQLClient)).then(({model, data}) => {
+      const promises = model.shop.products.reduce((promiseAcc, product, i) => {
+        const productData = data.shop.products.edges[i].node;
+
+        // Fetch the rest of the images for this product
+        if (productData.images && productData.images.pageInfo.hasNextPage) {
```

Comment:
What is the default page size for images and variants here? Just to avoid this being the common call path, can you request the max supported by the API (I think 250)?